src/kv/test/kv_dynamic_tables.cpp

```diff
@@ -331,6 +329,8 @@ TEST_CASE("Mixed map dependencies" * doctest::test_suite("dynamic"))
     REQUIRE(tx1.commit() == kv::CommitResult::SUCCESS);
     REQUIRE(tx2.commit() == kv::CommitResult::FAIL_CONFLICT);
 
+    tx2 = kv_store.create_tx();
```

Comment:
It seems odd to re-use a transaction here?

Reply:
This is deliberate to mimic the flow in `frontend.h` where the same name is re-used. Previously we'd call `tx2.reset()` inside the conflicting `commit()` on line 330, now we have to "manually reset" by assigning a freshly-constructed transaction. This is equivalent to saying `auto tx4 = ...` here, but by re-using `tx2` we also test the move-assignment that will be used in `frontend.h`.